predictsignauxfaibles/__main__.py

```diff
@@ -39,8 +51,31 @@ def load_conf(args):
     return model_conf
 
 
+def load_datasets_from_conf(args_ns, conf):
+    """
+    Configures train, test and predict dataset from user-provided options when pertaining.
+    """
+    datasets = {
+        "train": conf.TRAIN_DATASET,
+        "test": conf.TEST_DATASET,
+        "predict": conf.PREDICT_DATASET,
+    }
+
+    stats = {}
+
+    args_dict = vars(args_ns)
+    for (arg, dest) in args_to_attrs.items():
+        set_if_not_none(datasets[dest[0]], dest[1], args_dict[arg])
+        stats[arg] = getattr(datasets[dest[0]], dest[1])
+
+    if args_ns.predict_on is not None:
+        datasets["predict"].date_max = args_ns.predict_on[:-2] + "28"
+
+    return (datasets["train"], datasets["test"], datasets["predict"]), stats
```

Comment:
this function returns too many things, and nothing in its name or docstring suggests that I should expect to get a `stats` object

My suggestions : 
1. create the `stats` object in a separate function that is called on the 3 datasets _after_ they are done being modified
2. consider changing the name of the function to `get_train_test_predict_datasets()` ? So that the function return is clearer.

1 + 2 would result in something like this :

```python
train, test, predict = get_train_test_predict_datasets(args, conf)
stats = make_stats(train, test, predict)
```